### PR TITLE
Feature/confirmation

### PR DESCRIPTION
**Pull request by dominikx96**
I'm not sure what should be used as a "key" in FormSummaryDetails.tsx in maps:
- application.householdMembers
- accessibilityLabels(application.accessibility)

Maybe should we add unique IDs on the step when the user adding this information?

I also moved summary fragment to -> FormSummaryDetails.tsx, let me know if it's a good place for this kind of this component.

### CODE REVIEW

**Comment by jaredcwhite** [comment]
@dominikx96 Looks good! and nice work on extracting the summary information out into a reusable component.

FYI, I'd like to get #467 merged in before rebasing and merging this… also on the question regarding item keys, maybe @bencpeters can speak to that as he's been working on adding keys in other places.

**Comment by jaredcwhite** [comment]
@dominikx96 FYI, #467 got merged in

**Comment by bencpeters** [comment]
> I'm not sure what should be used as a "key" in FormSummaryDetails.tsx in maps:
> 
>     * application.householdMembers
> 
>     * accessibilityLabels(application.accessibility)
> 
> 
> Maybe should we add unique IDs on the step when the user adding this information?

For purposes of React `map`, keys are basically just used to help React build out its virtual DOM and keep track of each element in an array - it's not super important what they are, they just need to be unique.

For `householdMembers`, I'd probably just use `key=`${member.firstName} ${member.lastName}` - that should be unique unless the user makes a repeat entry.

For `accessibilityLabels`, the key should just be the `item` itself - the text should be unique. You'll have to use `<Fragment key={item}>`, since the `<>` shorthand doesn't support `key` props...

**Comment by dominikx96** [comment]
@slowbot so we should keep this component, but for now, #275 will be used in this step as default, right?

**Comment by bk3c** [comment]
@dominikx96 Correct - #275 is the screen that shows in response to a successful submission at the end of the flow. #276 is the screen with the details of the application that a user can log back in to see at a later time. 

Not part of this pair of issues, but the full flow is that once we have the functionality to send an email confirmation, that will include a link back to the site where the user can log in and then see the #276 screen.

**Comment by dominikx96** [comment]
@bk3c I updated this branch to be up to date with master. I also moved confirmation.tsx content to the new file: view-application.tsx.

A new confirmation.tsx is available in PR: #512 , so I think will be good to merge this first and next #512 .

**Comment by dominikx96** [comment]
@jaredcwhite I merged master into this branch and resolved conflicts. I think the flow is correct ...terms -> summary -> confirmation -> (optional view-application after clicking on "View submitted application and print a copy").

File: apps/public-reference/pages/applications/review/summary.tsx [hunk 1]
```diff
@@ -101,144 +48,7 @@ export default () => {
...
-            {application.incomeVouchers ? t("t.yes") : t("t.no")}
-          </ReviewItem>
-
-          <ReviewItem label={t("t.income")}>
-            ${application.income} {t(`application.financial.income.${application.incomePeriod}`)}
-          </ReviewItem>
-        </div>
+        <FormSummaryDetails application={application} editMode />
 
         <h3 className="form--card__sub-header">
```
Comment (jaredcwhite) [inline apps/public-reference/pages/applications/review/summary.tsx:53]: The new Preferences section should probably get moved into the FormSummaryDetails component as well.

File: apps/public-reference/pages/applications/review/view-application.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/*
```
Comment (jaredcwhite) [inline apps/public-reference/pages/applications/review/view-application.tsx:1]: Let's move this to something like /applications/view — I think we'll want to use it for referring to applications after time has passed, so it's not really part of the review flow.

File: apps/public-reference/pages/applications/review/summary.tsx (hunk 1 above)
Comment (dominikx96) [reply apps/public-reference/pages/applications/review/summary.tsx]: Absolutely yes! I think I missed this element after the merge.

File: apps/public-reference/pages/applications/review/view-application.tsx (hunk 2 above)
Comment (dominikx96) [reply apps/public-reference/pages/applications/review/view-application.tsx]: Sounds great, I changed the URL to /applications/view.

**Comment by jaredcwhite** [comment]
I think this is ready to merge now…can we get the QA sign off from @slowbot?

**Pull request merged**